File: 2022/day01/solutionB.rb
#!/bin/env ruby

input = File.read('./input.txt')

# @param {String} input
# @return {Integer[][]}
def format_input(input)
  input.split("\n\n").map do |numbers|
    numbers.split("\n").map(&:to_i)
  end
end

# @param {Integer[][]} formatted_input
# @return {Integer}
def solution(formatted_input)
  formatted_input.map(&:sum).sort.last(3).sum
end

print solution(format_input(input))

File: 2022/day01/python/solutionB.py
#!/bin/env python

input = open('../input.txt').read()

def format(input: str) -> 'map[map[int]]':
    return map(lambda n: map(int, n.split('\n')), input.split('\n\n'))

def solution(elves: 'map[map[int]]') -> int:
    return sum(sorted(map(sum, elves))[-3:])

print(solution(format(input)))
